misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -2969,6 +2929,134 @@ public BreakStatementNode transform(BreakStatementNode breakStatementNode) {
                 .apply();
     }
 
+    @Override
+    public NamedWorkerDeclarationNode transform(NamedWorkerDeclarationNode namedWorkerDeclarationNode) {
+        NodeList<AnnotationNode> annotations = formatNodeList(namedWorkerDeclarationNode.annotations(), 0, 0, 0, 0);
+        Token workerKeyword = formatToken(namedWorkerDeclarationNode.workerKeyword(), 1, 0);
+        IdentifierToken workerName = formatToken(namedWorkerDeclarationNode.workerName(), 1, 0);
+        if (namedWorkerDeclarationNode.returnTypeDesc().isPresent()) {
+            Node returnTypeDesc = formatNode(namedWorkerDeclarationNode.returnTypeDesc().get(), 0, 0);
+            namedWorkerDeclarationNode = namedWorkerDeclarationNode.modify()
+                    .withReturnTypeDesc(returnTypeDesc)
+                    .apply();
+        }
+
+        BlockStatementNode workerBody = formatNode(namedWorkerDeclarationNode.workerBody(), 0, 1);
+        
+        return namedWorkerDeclarationNode.modify()
+                .withAnnotations(annotations)
+                .withWorkerKeyword(workerKeyword)
+                .withWorkerName(workerName)
+                .withWorkerBody(workerBody)
+                .apply();
+    }
+
+    @Override
+    public ArrayTypeDescriptorNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
+        TypeDescriptorNode memberTypeDesc = formatNode(arrayTypeDescriptorNode.memberTypeDesc(), 0, 0);
+        Token openBracket = formatToken(arrayTypeDescriptorNode.openBracket(), 0, 0);
+        Token closeBracket = formatToken(arrayTypeDescriptorNode.closeBracket(), this.trailingWS, 0);
+        if (arrayTypeDescriptorNode.arrayLength().isPresent()) {
+            Node arrayLength = formatNode(arrayTypeDescriptorNode.arrayLength().orElse(null), 0, 0);
+            arrayTypeDescriptorNode = arrayTypeDescriptorNode.modify()
+                    .withArrayLength(arrayLength)
+                    .apply();
+        }
+
+        return arrayTypeDescriptorNode.modify()
+                .withOpenBracket(openBracket)
+                .withCloseBracket(closeBracket)
+                .withMemberTypeDesc(memberTypeDesc)
+                .apply();
+
+    }
+
+    @Override
+    public XMLElementNode transform(XMLElementNode xMLElementNode) {
+        XMLStartTagNode startTagNode = formatNode(xMLElementNode.startTag(), 0, 0);
+        NodeList<XMLItemNode> content = formatNodeList(xMLElementNode.content(), 0, 0, 0, 0);
+        XMLEndTagNode endTagNode = formatNode(xMLElementNode.endTag(), 0, 0);
```

Comment:
Last child in all nodes (i.e: in all the transform methods), should take the trailing WS/NL from the global variable.

Lets fix that in all places

Reply:
Fixed.